analyses/mwi/dryspells_trigger/exploration/2021_2022_postseason_overview.Rmd

```diff
@@ -0,0 +1,570 @@
+
+```{r duration-plot}
+plot(static_r, 
+     "duration", 
+     type = "interval", 
+     breaks = duration_breaks, 
+     main = "Rainy Season Duration",
+     axes = F,
+     plg = list(title = "In Days"),
+     col = duration_palette[1:(length(duration_breaks) - 1)])
+plot(southern_shp, add = T)
+
+```
+
+```{r q1}
+perc_nov_onset <- season_dates %>% filter(onset_date < '2021-12-01') %>% summarise(nov_onset = as.numeric(n_distinct(cell))) %>% mutate(round(100 * nov_onset / nbr_cells, 0)) %>% select(-nov_onset) %>% as.numeric()
+
+onset_90perc_known <- as.numeric(quantile(season_dates$onset_days_since_1nov, 0.9))
+onset_50perc_known <- as.numeric(quantile(season_dates$onset_days_since_1nov, 0.5))
+```
+
+## On Rainfall Patterns
+This section examines rainfall patterns.
+
+### Rainy Days
+[DCCMS](https://www.metmalawi.gov.mw/dccms_agromet.php) in its latest bulletins defines "rainy day" has having received at least 3mm of rain. This definition was adopted for this analysis.
+
+The graph on the left (next section below) shows the ratio of rainy days over season duration (ie., as a percentage of days). The yellow-pale green areas received rain on a smaller portion of days compared to the teal-blue areas. Notably, no area received rain on more than `r minmax(static_r[["rainy_ratio"]])[2]`% of days during the rainy season. Thus, dry days were quite common (up to `r 100 - minmax(static_r[["rainy_ratio"]])[1]`% of days in areas with the least rainy days).
+
+### Rainfall Amounts
+The cumulative amount of rainfall over the duration of the rainy season was computed and is shown on the right below. Darker colours indicate highest rainfall totals. 
+
+There were large variations in amount of rainfall received. Although there is a relation between total rainfall and season duration, and between total rainfall and ratio of dry days, it is not a straightforward one: not all areas with the longest durations or highest ratios of rainy days received the most rain.
+
+
+```{r rainy-days-plots}
+
+rainy_ratio_labels <- c( "0-20", "21-40", "41-60", "61-80", "81-100")
+  
+rain_seas_tot_min <- plyr::round_any(minmax(static_r[["rain_seas_tot"]])[1], 100, f=floor)
+rain_seas_tot_max <- plyr::round_any(minmax(static_r[["rain_seas_tot"]])[2], 100, f=ceiling)
+rain_seas_tot_breaks <- seq(from = plyr::round_any(rain_seas_tot_min, 100, f=floor), 
+                            to = plyr::round_any(rain_seas_tot_max, 100, f=ceiling), # ceiling ensures max value included
+                            by = 100)
+
+par(mfrow = c(1,2),
+    mar = c(1, 1, 3, 1)) # bottom, left, top, right.
+
+plot(static_r, 
```

Comment:
I like this, I think my concern is that since each cell has their own rainy period duration, the comparisons are not so meaningful. Those with shorter rainy seasons have higher percents. Could the plot instead just be from the frist potential start of the rainy season and the first potential end, for instance? Or the latest date of any rainy season? So that you can compare % of rainy days across each. Then you could see, if for instance, there's significant difference between the areas with a later rainy season start than those with an earlier in terms of actual rainy days, or maybe it was just very small differences.